test/contrib/emscripten/test_emscripten.py

```diff
@@ -0,0 +1,381 @@
+from __future__ import annotations
+
+import typing
+
+import pytest
+
+from urllib3.fields import _TYPE_FIELD_VALUE_TUPLE
+
+# only run these tests if pytest_pyodide is installed
+# so we don't break non-emscripten pytest running
+pytest_pyodide = pytest.importorskip("pytest_pyodide")
+
+from pytest_pyodide import run_in_pyodide  # type: ignore[import] # noqa: E402
+from pytest_pyodide.decorator import (  # type: ignore[import] # noqa: E402
+    copy_files_to_pyodide,
+)
+
+from .conftest import PyodideServerInfo, ServerRunnerInfo  # noqa: E402
+
+# make our ssl certificates work in chrome
+pytest_pyodide.runner.CHROME_FLAGS.append("ignore-certificate-errors")
+
+
+@copy_files_to_pyodide(file_list=[("dist/*.whl", "/tmp")], install_wheels=True)  # type: ignore[misc]
+def test_index(selenium: typing.Any, testserver_http: PyodideServerInfo) -> None:
+    @run_in_pyodide  # type: ignore[misc]
```

Comment:
Is it possible to turn `@run_in_pyodide` into a pytest fixture that executes the pytest function itself (and injects selenium, testserver, etc?)

Reply:
You can often use it like that, but I couldn't get it to work here - it requires all the arguments to be simple and pickleable, and there are other type constraints. I haven't delved into it as @run_in_pyodide is non-trivial and I don't want to mess with it when using an inner function with basic types is functional.